Update README to synergize with docs (#253)
neo4j/mcp: docs/TLS_SETUP.md

Reviewer: I think there's still some value in this guide in the context of docs for contributors

I'd probably just cut the scope slightly to not minimise descriptions of config options and to make clear this guide is for contributors manually testing the server, not for real users/deployments of the server

Author: restored the file and added a note -
for reducing the scope, what should go?

Reviewer: Just had another read through and I think let's just leave this as it is for the most part. I'll put some specific comments

Reviewer: If we cut the scope to just self signed certs for manual testing this isn't needed

```suggestion
```

Reviewer: Not sure if the suggested change is working for some reason 🤔

# TLS/HTTPS setup for Neo4j MCP server

> This guide covers TLS/HTTPS configuration for the Neo4j MCP server, including certificate generation, testing, and production deployment.

Reviewer: ```suggestion
This guide covers TLS/HTTPS configuration for contributors who are manually testing the Neo4j MCP server during development. It uses self-signed certificates which are not suitable for production deployments.
```

> **Note**: This guide is meant for contributors who are manually testing the server, not for users and deployments of the server.

Reviewer: If you use my suggestion above to set the scope of this guide, this is covered

```suggestion
```

## Important certificate requirements

### Certificate format

> All certificates must be in **PEM format** (text-based format with `-----BEGIN CERTIFICATE-----` headers). The server does not support other formats like DER or PKCS12.

Reviewer: The main docs should be the source of truth for this

```suggestion
```

# TLS/HTTPS setup for Neo4j MCP server

This guide covers TLS/HTTPS configuration for contributors who are manually testing the Neo4j MCP server during development. It uses self-signed certificates which are not suitable for production deployments.

## Important certificate requirements

### Certificate authority

**Self-Signed Certificates**: Self-signed certificates do not work out of the box with many MCP clients (e.g., VSCode Copilot, Claude Desktop). These clients require certificates signed by a trusted Certificate Authority (CA).

> See the [Production Use](#production-use) section below for proper setup.

Reviewer: Can we remove anything about production use from this guide

```suggestion
```

Author: removed

See the [Production Use](#production-use) section below for proper setup.

**Note**: Automated tests generate certificates dynamically.

> For manual testing or production deployment, follow the steps below.

Reviewer: This guide should only be for manual testing

```suggestion
```

./bin/neo4j-mcp

> ```

Reviewer: To keep it simple let's just do the one example 

```suggestion
```

Reviewer: Again, not sure the suggested change is working...

Author: just tested, i think it's because of the three backticks

- **Content-Type**: MCP requests need `Content-Type: application/json` header
- **Port**: Default port is 443 when TLS is enabled, 80 when TLS is disabled (configurable via `--neo4j-http-port` or `NEO4J_MCP_HTTP_PORT`)

> ## Production use

Reviewer: Can we remove the production use section from this guide

Author: removed

# Other paths will return 404
curl -u neo4j:password https://your-domain.com/
# Returns: "Not Found: This server only handles requests to /mcp"

> ```

Reviewer: This guide shouldn't contain any mention of production use if it's for internal contributors
